Assets/Scripts/PauseMenu.cs

```diff
@@ -0,0 +1,107 @@
+using UnityEngine;
+using UnityEngine.Audio;
+using UnityEngine.SceneManagement;
+using UnityEngine.UI;
+using Yarn.Unity;
+
+public class PauseMenu : MonoBehaviour
+{
+    public static bool GameIsPaused;
+    [SerializeField] private GameObject pauseMenuUI;
+
+    [SerializeField] private AudioMixer audioMixer;
+
+    [SerializeField] private Slider BGMSlider;
+    [SerializeField] private Slider SFXSlider;
+
+    [SerializeField] private float startBGMVolume = 0.5f;
+    [SerializeField] private float startSFXVolume = 0.5f;
+    [SerializeField] private float startMasterVolume = 0.5f;
+
+    // SerializeField to directly assign the AudioSource that plays the SFX in the Inspector
+    [SerializeField] private AudioSource sfxAudioSource;
+    private DialogueRunner _dialogueRunner;
+
+    public void Start()
+    {
+        pauseMenuUI.SetActive(false);
+        audioMixer.SetFloat("Master", startMasterVolume);
+        var savedBGMVolume = PlayerPrefs.GetFloat("BGM", startBGMVolume);
+        var savedSFXVolume = PlayerPrefs.GetFloat("SFX", startSFXVolume);
+        SetBGMVolume(savedBGMVolume);
+        SetSFXVolume(savedSFXVolume);
+        BGMSlider.value = savedBGMVolume;
+        SFXSlider.value = savedSFXVolume;
+
+        _dialogueRunner = FindObjectOfType<DialogueRunner>();
+        Debug.Log(_dialogueRunner);
+        // _dialogueRunner = GetComponent<DialogueRunner>();
+        // Debug.Log(_dialogueRunner);
+        Resume();
+    }
+
+    // Update is called once per frame
+    private void Update()
+    {
+        if (Input.GetKeyDown(KeyCode.Escape))
+        {
+            if (GameIsPaused)
+                Resume();
+            else
+                Pause();
+        }
+    }
+
+    public void SetBGMVolume(float volume)
```

Comment:
This looks duplicated from MainMenu, I would suggest moving them to SoundManager